fl/src/datasets/memory.py

```diff
@@ -3,44 +3,53 @@
 from pgenlib import PgenReader
 
 
-def load_from_pgen(pfile_path: str, gwas_path: str, snp_count: int) -> numpy.ndarray:
+def load_from_pgen(pfile_path: str, gwas_path: str, snp_count: int, missing='zero') -> numpy.ndarray:
     """
     Loads genotypes from .pgen into numpy array and selects top {snp_count} snps
 
     Args:
         pfile_path (str): Path to plink 2.0 .pgen, .pvar, .psam dataset. It should not have a .pgen extension.
         gwas_path (str): Path to plink 2.0 GWAS results file generated by plink 2.0 --glm. 
-        snp_count (int): Number of most significant SNPs to load 
+        snp_count (int): Number of most significant SNPs to load. If None then load all SNPs
+        missing (str): Strategy of filling missing values. Default is 'zero', i.e. homozygous reference value. Other is 'mean'.
 
     Raises:
         ValueError: If snp_count is greated than number of SNPs in .pgen
 
     Returns:
         numpy.ndarray: An int8 sample-major array with {snp_count} genotypes
     """    
-    reader = PgenReader(pfile_path + '.pgen'.encode('utf-8'))
+    reader = PgenReader((pfile_path + '.pgen').encode('utf-8'))
```

Comment:
At FL step we can use up to {k_max} top SNPs. Job union.py produces pgen files with {k_max} top SNPs and load_from_pgen can read any arbitrary number of SNPs from 1 to {k_max}. With this approach we can perform experiments on any number of SNPs from 1 to {k_max}, save space (one pgen with {k_max} SNPs instead of several files with {k_1, k_2,...,k_max} SNPs), and load data faster than if we load it from source pgen with 700K SNPs.